admin/scripts/common/models/__init__.py

```diff
@@ -0,0 +1,105 @@
+from enum import Enum
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field, field_validator, model_validator
+
+
+class Choice(str, Enum):
+    yes = "Yes"
+    not_planned = "Not planned"
+    not_yet = "Not yet"
+
+
+class Choice2(str, Enum):
+    yes = "Yes"
+    not_sure_yet = "Not sure yet"
+    no = "No"
+
+
+class ChoiceReminder(str, Enum):
+    no = "No"
+    yes_one_week = "Yes, in one week"
+    yes_one_month = "Yes, in one month"
+
+
+class ProjectRole(str, Enum):
+    principal_author_lead_engineer = "Principal author / lead engineer"
+    core_contributor_maintainer = "Core contributor / maintainer"
+    release_manager = "Release manager"
+    security = "Security"
+    infrastructure_devops = "Infrastructure/DevOps"
+    community_manager = "Community manager"
+    outreach_coordinator = "Outreach coordinator"
+    other = "Other"
+
+
+class UpdateFrequency(str, Enum):
+    cant_say = "Can't say"
+    more_once_per_day = "More than once per day"
+    more_once_per_week = "More than once per week"
+    more_once_per_month = "More than once per month"
+    more_once_per_year = "More than once per year"
+    less_once_per_year = "Less than once per year"
+
+
+question_alias_mapping = {
+    "q1": "project_name",
+    "q2": "author_role",
+    "q3": "build_failure_duration",
+    "q4": "automatic_dependency_update",
+    "q5": "dependency_update_frequency",
+    "q6": "contributors",
+    "q24": "structured_data_provided",
+    "q25": "reminder",
+}
+
+
+class Form(BaseModel):
+    class Response(BaseModel):
+        @model_validator(mode="before")
+        def map_aliases(cls, values: dict[str, Any]) -> dict[str, Any]:
+            for alias, field_name in question_alias_mapping.items():
+                if alias in values:
+                    values[field_name] = values.pop(alias)
+            return values
+
+        @field_validator(question_alias_mapping["q24"], mode="before")
+        def map_yes_no_to_bool(cls, value: str) -> bool:
+            if value.lower() == "yes":
+                return True
+            return False
+
+        @field_validator(question_alias_mapping["q3"], mode="before")
+        def map_duration(cls, value: str) -> str:
+            if value.replace(".", "", 1).isdigit():
+                return f"{value}h per week"
+            return value
```

Comment:
Does it make more sense the map values here (from the information source) or in the other data classes? For example, a `Project` might need the duration to be in a specific form, while `Notion` might require it to be in another. In that case, it would probably be better for field validation to happen separately.